packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js

```diff
@@ -2109,22 +2099,164 @@ describe('Checkout', () => {
                     }
                 });
 
-                const payload = {
-                    action: 'submit'
-                };
+                // Act & Assert
+                expect(wrapper.vm.onFormSubmit).toBeDefined();
+            });
 
-                // Act
-                await wrapper.vm.onFormSubmit();
+            describe('when invoked', () => {
+                it('should make a call to `trackFormInteraction` so we can track the action type `submit`', async () => {
+                    // Arrange
+                    isFormValidSpy.mockReturnValue(true);
+                    const trackFormInteractionSpy = jest.spyOn(VueCheckout.methods, 'trackFormInteraction');
+                    const wrapper = mount(VueCheckout, {
```

Comment:
Maybe not one for this PR but you could reduce duplication when setting up the component by creating a top-level variable that sets all of the defaults for you whilst allowing you to pass custom variables e.g.

```js
const createComponent = ({
	store = createStore()
} = {}) => mount(VueCheckout, {
	  store,
	  i18n,
	  localVue,
	  propsData,
	  mocks: {
	      $v,
	      $logger
	  }
});

```

Reply:
Sounds good, there's still a bit of a refactor for one other change so i will look at doing this in the next one.